kapitan/secrets.py

```diff
@@ -433,6 +432,24 @@ def rsa_private_key(key_size=''):
     ), "UTF-8")
 
 
+def rsa_public_key(secrets_path, private_key_file):
+    """derives an RSA public key from private_key_file"""
+    token = "gpg:%s" % private_key_file
+    secret_raw_obj = secret_gpg_raw_read(secrets_path, token)
+    data_dec = secret_gpg_read(secrets_path, token)
+    encode_base64 = secret_raw_obj.get('encoding', None) == 'base64'
+
+    if encode_base64:
+        data_dec = base64.b64decode(data_dec).decode('UTF-8')
+
+    private_key = serialization.load_pem_private_key(data_dec.encode(), password=None, backend=default_backend())
+
+    return str(private_key.public_key().public_bytes(
```

Comment:
Nit: IMO it would be easier to read if you extract generating public key to its own variable.